src/Microsoft.OpenApi/Interfaces/IOpenApiReader.cs

```diff
@@ -0,0 +1,121 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. 
+
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.OpenApi.Models;
+using Microsoft.OpenApi.Reader;
+
+namespace Microsoft.OpenApi.Interfaces
+{
+    /// <summary>
+    /// Interface for Open API readers.
+    /// </summary>
+    public interface IOpenApiReader
+    {
+        /// <summary>
+        /// Reads the input URL and parses it into an Open API document.
+        /// </summary>
+        /// <param name="url">The input to read from.</param>
+        /// <param name="diagnostic">The diagnostic entity containing information from the reading process.</param>
+        /// <param name="settings"> The OpenApi reader settings.</param>
+        /// <returns></returns>
+        OpenApiDocument Read(string url, out OpenApiDiagnostic diagnostic, OpenApiReaderSettings settings = null);
```

Comment:
This interface doesn't need all of these options.  The user will never call these.  We only need the most primitive one. e.g. TextReader